projects/baby_driver/scripts/test_spi_exchange.py

```diff
@@ -0,0 +1,141 @@
+        self.data = None
+        self.channel = None
+        self.msg_id = None
+        self.device_id = None
+
+        def send_msg_test(
+            babydriver_id=None,
+            data=None,
+            channel=None,
+            msg_id=BABYDRIVER_CAN_MESSAGE_ID,
+            device_id=BABYDRIVER_DEVICE_ID,
+        ):
+            self.babydriver_id = babydriver_id
+            self.data = data
+            self.channel = channel
+            self.msg_id = msg_id
+            self.device_id = device_id
+            self.return_rx_data = [BabydriverMessageId.SPI_EXCHANGE_RX_DATA, 0, 0, 0, 0, 0, 0, 0]
+            self.rx_status = [BabydriverMessageId.SPI_EXCHANGE_TX_DATA, OK_STATUS]
+
+        mock_send_message.side_effect = send_msg_test
+
+        data_msg = [BabydriverMessageId.SPI_EXCHANGE_RX_DATA, 0, 0, 0, 0, 0, 0, 0]
+        status_msg = [BabydriverMessageId.STATUS, OK_STATUS]
+        mock_next_message.side_effect = (Message(data=data_msg), Message(data=status_msg))
+
+        # Normal test
+        self.assertEqual(spi_exchange(
+            tx_bytes=[1, 1, 1, 1, 1, 1, 1, 1, 1, 1],  # 10 bits
+            rx_len=5,
+            spi_port="B",
+            spi_mode=0,
+            baudrate=5000000,
+            cs=("A", 1),
+        ), [0, 0, 0, 0, 0])
+
+        # Test with spi_port as an int value
+        mock_next_message.side_effect = (Message(data=data_msg), Message(data=status_msg))
+        self.assertEqual(spi_exchange(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 5, 1, 0, 5000000, (1, 1)
+            ), [0, 0, 0, 0, 0]
+        )
+
+        # Test default values
+        mock_next_message.side_effect = (Message(data=data_msg), Message(data=status_msg))
+        self.assertEqual(spi_exchange(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 5
+        ), [0, 0, 0, 0, 0])
+
+        # 0 rx_len
+        mock_next_message.side_effect = (Message(data=data_msg), Message(data=status_msg))
+        self.assertEqual(spi_exchange(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 0, "B", 0, 5000000, ("A", 1),
+        ), [])
+
+        # Test rx_len < 7
+        mock_next_message.side_effect = (Message(data=data_msg), Message(data=status_msg))
+        self.assertEqual(spi_exchange(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 6
+        ), [0, 0, 0, 0, 0, 0])
+
+        # Test rx_len equal a multiple of 7
+        mock_next_message.side_effect = (Message(data=data_msg), Message(data=status_msg))
+        self.assertEqual(spi_exchange(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 7
+        ), [0, 0, 0, 0, 0, 0, 0])
+
+        # Test rx_len > 7
+        # pylint: disable=line-too-long
+        mock_next_message.side_effect = (Message(data=data_msg), Message(data=data_msg), Message(data=status_msg))
+        self.assertEqual(spi_exchange(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 8
+        ), [0, 0, 0, 0, 0, 0, 0, 0])
+
+        # Test multiple messages, low rx_len
+        # pylint: disable=line-too-long
```

Comment:
If the lines are too long, then wrap the line rather than just disabling the message!